lsp/lsp.go

```diff
@@ -149,3 +173,36 @@ func setTrace(context *glsp.Context, params *protocol.SetTraceParams) error {
 	protocol.SetTraceValue(params.Value)
 	return nil
 }
+
+// getLineOrWordLength returns the length of the line or the length of the word starting at the given column.
+// If wholeLine is true, it returns the length of the entire line.
+// If wholeLine is false, it returns the length of the word starting at the column.
+func getLineOrWordLength(filePath string, lineNum, column int, wholeLine bool) (int, error) {
```

Comment:
While this is probably okay as a stop-gap, we should really be extending the error type to include range information so we can include it when the errors are created, as this is incredibly inefficient.

Reply:
I started down this path and got pretty hung up on parser stuff (I think that was the source of the issue). It was a bit of a time suck so I just put this in for now to keep moving forward.

I suspect it had something to do with how our `schema.Position` doesn't exactly match `participle.Position` but I'm fully just guessing. Once I added `EndColumn` to `schema.Position` I lost all position information. I do have a branch with that work though we can pick up when you have a sec :)

Reply:
You assumption is correct: the Position type is a mirror of participles so you'll need to either add the end location to the error rather than the position, or include it in our position type and write a function to do the mapping.